new_dawn_server/users/api/resources.py

```diff
@@ -158,15 +160,62 @@ def phone_verify_authenticate(self, request, **kwargs):
                 request, {"success": False, "message": "Missing phone_number or country_code"}, HttpNoContent)
 
 
+class LocationValidation(Validation):
+    def is_valid(self, bundle, request=None):
+        if not bundle.data.get("city_preference"):
+            return {'__all__': "No data for city_preference"}
+
+        if not isinstance(bundle.data.get("city_preference"), list):
+            return {'__all__': "city_preference is not a list"}
+
+
 class AccountResource(ModelResource):
-    user = fields.ToOneField(UserResource, "user", related_name="account", full=True)
 
     class Meta:
-        allowed_methods = ["get"]
+        allowed_methods = ["get", "post"]
+        always_return_data = True
         authentication = Authentication()
         authorization = Authorization()
-        queryset = Account.objects.all()
+        queryset = User.objects.all()
         resource_name = "account"
+        validation = LocationValidation()
+
+    @staticmethod
+    def _get_account_name(first_name, last_name):
+        return first_name + ACCOUNT_NAME_DELIMITER + last_name
+
+    @staticmethod
+    def _get_model_fields_dict(bundle, fields):
+        result_dict = {}
+        for field, _ in fields.items():
+            # The field can be None if it's not in the bundle
+            result_dict[field] = bundle.data.get(field)
+        return result_dict
+
+    def obj_create(self, bundle, **kwargs):
```

Comment:
Just merge these part into UserRegistrationResource.

Reply:
So you want city preference to be co-created with user registration?

Reply:
Yes let's do that for now.
If user want to change city preference, he can always use "update" API from tastypie. We will expand more on that later when we develop setting page.